### DISCUSSION TITLE
FR: Relative expiry dates

### DISCUSSION THREAD

**Discussion by ldstevens** [engram-design/FeedMe#875] [Ideas]
I'm guessing this isn't possible right now, but it would be handy to be able to set relative expiry dates for imported items (e.g. 30 days/2 months/1 year from time of import). I'm keen to use relative expiry dates for a side project where aggregated content (RSS items) will pass through Craft. Cheers ✌️

**Comment by ldstevens** [comment]
["Allow Twig shorthand variables to be used in default fields"](https://github.com/engram-design/FeedMe/commit/a761f89a06b232ab178b29ea2558647b728ab9e2) sounds really cool. Would this allow you to use Twig's `now` with an offset in the default field for relative expiry dates? (Been meaning to make a FR for that exact functionality!) Or have I misunderstood?

**Comment by engram-design** [comment]
Thats correct, although note the use of full Twig syntax using `{{now}}`. You can either use this in the default field value - or have this twig directly in your feed (if you have control over that).

For instance:
`<Date2>{{now}}</Date2>`

**Comment by ldstevens** [comment]
Awesome.

One small issue I noticed in the logs - it appears {whatever} is getting treated as a Twig variable in content fields:

> News import: An exception has been thrown during the rendering of a template ("Craft\EntryModel and its behaviors do not have a method or closure named "filedir_5".") in `"<p><img alt="" src="{{object.filedir_5|raw}}samyang-35mm-t1.3-camera-lens-specification_.jpg" style="width: 550px; height: 381px;" /></p>"` at line 1.

I think the actual field value is `src="{filedir_5}samyang-35mm-t1.3-camera-lens-specification_.jpg"`. (It's an EE thing.)

**Comment by engram-design** [comment]
Yeah, thats a bit of a tricky one, as thats essentially a Twig error. If you specify a field handle that doesn't exist `{myFieldHandle}`, the Twig parser will complain that it doesn't exist on the object.

I've actually just improved this a little in the latest master.
